cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -817,6 +817,10 @@ private void emitRunTimeMetric(ProgramRunId programRunId, ProgramRunStatus progr
     MetricsContext metricsContext = ProgramRunners.createProgramMetricsContext(programRunId, tags,
                                                                                metricsCollectionService);
     metricsContext.gauge(Constants.Metrics.Program.RUN_TIME_SECONDS, runTime);
+    metricsContext.gauge(Constants.Metrics.Program.EXPECTED_STOP_TIME_SECONDS,
+                         Optional.ofNullable(runRecord.getTerminateTs()).orElse(-1L));
+    metricsContext.gauge(Constants.Metrics.Program.ACTUAL_STOP_TIME_SECONDS,
+                         Optional.ofNullable(runRecord.getStopTs()).orElse(-1L));
```

Comment:
similar comment here, this is the timestamp for when the run completed. What we want to emit is the time elapsed from stopping to actually being stopped, not the timestamps for when they happened.